### PR TITLE
samples: refactor Bigtable tests using a base class

### PR DESCRIPTION
**Pull request by billyjacobson**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: samples/snippets/src/test/java/com/example/bigtable/BigtableBaseTest.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+import java.io.IOException;
+import java.io.PrintStream;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Random;
+import java.util.UUID;
+import org.junit.After;
+import org.junit.Before;
+
+public class BigtableBaseTest {
```
Comment (kolea2) [inline samples/snippets/src/test/java/com/example/bigtable/BigtableBaseTest.java:39]: is this class actually running any tests? If no, I would rename this as a Helper class, and instead have the tests not extend it but statically import and use these helper methods.

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: samples/snippets/src/test/java/com/example/bigtable/BigtableBaseTest.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  // Environment variables
+  public static String projectId;
+  public static String instanceId;
+
+  // Test output stream
+  private static PrintStream originalOut;
+  public ByteArrayOutputStream bout;
+
+  // Mobile time series db variables
+  public static final String TABLE_ID = generateTableId("mobile-time-series");
```
Comment (kolea2) [inline samples/snippets/src/test/java/com/example/bigtable/BigtableBaseTest.java:50]: I think this class could be broken up a bit. The generic stuff that happens in each test (`@Before` setup, `@After` teardown, etc) should remain here. However, the mobile time series variables and methods are not relevant to every sample and are a bit confusing to have available in things like the channel pooling example. I would take all of that and put it into a static helper class to be called by those tests instead.

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: samples/snippets/src/test/java/com/example/bigtable/BigtableBaseTest.java (hunk 2 above)
Comment (billyjacobson) [reply samples/snippets/src/test/java/com/example/bigtable/BigtableBaseTest.java]: done

**Review by kolea2 — APPROVED** [review]
Looks awesome! Thanks for doing this

**Pull request merged**